### ISSUE TITLE
Error when using AppSignal 3 with existing pg library and OpenTelemetry instrumentation

### ISSUE THREAD

**Issue by evenfrost** [appsignal/appsignal-nodejs#810]
I upgraded from v2.x to v3 following the migration guide, and now I'm getting the following error when trying to start the application:

```
> node --require ./build/server/appsignal.cjs ./build/server/index.js

/home/project/node_modules/knex/lib/dialects/postgres/index.js:115
        connection.on('error', (err) => {
                   ^

TypeError: Cannot read properties of undefined (reading 'on')
    at Client.<anonymous> (/home/project/node_modules/knex/lib/dialects/postgres/index.js:115:20)
    at Client.patchedClientConnectCallback [as _connectionCallback] (/home/project/node_modules/@opentelemetry/instrumentation-pg/build/src/utils.js:187:12)
    at Client._handleReadyForQuery (/home/project/node_modules/pg/lib/client.js:279:14)
    at Connection.emit (node:events:527:28)
    at Connection.emit (node:domain:475:12)
    at /home/project/node_modules/pg/lib/connection.js:114:12
    at Parser.parse (/home/project/node_modules/pg-protocol/dist/parser.js:40:17)
    at Socket.<anonymous> (/home/project/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:527:28)
    at Socket.emit (node:domain:475:12)
```

It seems that `@opentelemetry` patches some method, which is being used by Knex afterwards, and this gives an error (possible due to clash of scopes/names because of the patching, but this is just an assumption).

**Comment by tombruijn** [comment]
Hi @evenfrost, this hasn't happened in our example app also using PostgreSQL so we'll need to reproduce.

What version of knex are you using?
Can you share some example code that triggers this error?

To work around it, you can configure [disableDefaultInstrumentations](https://docs.appsignal.com/nodejs/3.x/configuration/options.html#option-disabledefaultinstrumentations) with `@opentelemetry/instrumentation-knex` to disable the instrumentation and make the app work again.

**Comment by evenfrost** [comment]
Here is a minimal repro repository: https://github.com/evenfrost/appsignal-knex-issue.

A bit of backstory: we're using [`mikro-orm`](https://github.com/mikro-orm/mikro-orm) as our Postgres ORM, which uses `knex` internally for SQL management, and faced this issue with `v4` of `mikro-orm`. However, a quick upgrade to `v5` in this demo repository showed that the error was gone, possibly due to an upgrade of underlying libraries. The problem is that upgrading our infrastructure to `v5` will take plenty of work, so I'd like to know if there is a solution to this issue that doesn't require it.

**Comment by tombruijn** [comment]
How do I trigger the error as described in this issue? If I start the app it doesn't do much, but also doesn't error.

We'll have a look if we can support v4 easily by fixing the issue, but we can't promise anything.

**Comment by evenfrost** [comment]
You need to provide existing Postgres connection parameters along these lines: https://github.com/evenfrost/appsignal-knex-issue/blob/master/src/db.ts#L24.

Also, the error is gone with the `disableDefaultInstrumentations: ['@opentelemetry/instrumentation-pg']` option, so I believe it's more of `pg` than `knex` thing, though. But if I disable `pg` instrumentation, will it have side effects like not logging SQL queries in AppSignal?

**Comment by tombruijn** [comment]
> You need to provide existing Postgres connection parameters along these lines>

Okay, I now have the error.

> Also, the error is gone with the disableDefaultInstrumentations: ['@opentelemetry/instrumentation-pg'] option, so I believe it's more of pg than knex thing, though.

I realized that as well later.

> if I disable pg instrumentation, will it have side effects like not logging SQL queries in AppSignal?

Yes, it will completely disable the pg instrumentation and not report any queries.

**Comment by tombruijn** [comment]
Looks like this issue, reported on the OpenTelemetry repo, is the same one or at least related: https://github.com/open-telemetry/opentelemetry-js-contrib/issues/991
Some people reported downgrading to `@opentelemetry/instrumentation-pg0.31.1` may help, but I can't get it to work.

**Comment by unflxw** [comment]
The issue (https://github.com/open-telemetry/opentelemetry-js-contrib/issues/991) was fixed, and the fix was released in version 0.35.0 of the `@opentelemetry/instrumentation-pg` package.

AppSignal for Node.js version 3.0.13 was just released, updating OpenTelemetry dependencies to the latest versions, including the release of the PostgreSQL instrumentation with the fix. I have also verified that the issue reported here no longer happens in the test setup provided (thanks @evenfrost!)
